src/lib/cloud-sdk-api.js

```diff
@@ -228,14 +228,22 @@ class CloudSdkAPI {
       const url = result.headers.get('Location');
       const changeId = (await result.json()).updateId;
       const client = new ShareFileClient(url);
-      uploadCallbacks.start(fileSize);
+      if (uploadCallbacks) {
+        uploadCallbacks.start(fileSize);
+      }
```

Comment:
Isn't this equivalent to the following, which is shorter?

```suggestion
        uploadCallbacks?.start(fileSize);
```

Alternatively, you could consider passing in no-op callbacks, in case that makes the code-flow simpler. I have no preference in this regard though.

Reply:
yes it is, I stuck with java mindset still =)